lib/vmpooler/pool_manager.rb

```diff
@@ -459,69 +458,17 @@ def _check_snapshot_queue
       end
     end
 
-    def migration_limit(migration_limit)
-      # Returns migration_limit setting when enabled
-      return false if migration_limit == 0 || !migration_limit # rubocop:disable Style/NumericPredicate
-      migration_limit if migration_limit >= 1
-    end
-
     def migrate_vm(vm_name, pool_name, provider)
       Thread.new do
         begin
```

Comment:
The provider really shouldn't need to contact redis for this.  In this case the provider is just removing it from the queue.  So the `$redis.srem('vmpooler__migrating__' + pool_name, vm_name)` should be called here PRIOR to the `provider.migrate_vm` call

Reply:
I took your suggestion and placed this prior to calling `provider.migrate_vm`.